sdk/python/test_data/components/container_with_if_placeholder.yaml

```diff
@@ -9,6 +9,7 @@ components:
       parameters:
         optional_input:
           defaultValue: default
+          isOptional: true
```

Comment:
It seems like all the test golden files updates are for optional parameter inputs, which isn't really improving support but fixing a bug: information lost for optional input parameter with None default value when preserving into YAML.
It's fine to go ahead with this change. But ideally, in the future we should keep the changes separate: fixing bug for optional input parameters, and add support for optional input artifacts.